### PR TITLE
refactor: localization abstract base class

### PR DESCRIPTION
**Pull request by hayato-m126**
## Types of PR

- [x] Upgrade of existing features

## Description

- Common functions of the evaluator_node are integrated in driving_log_replayer/evaluator.py
- Each evaluation node inherits DLREvaluator of evaluator and implements it.
- Each evaluation node writes pub/sub, callback function, and result code for evaluation.
- Set ResultBase in driving_log_replayer/result.py to ABC and set the function to abstractmethod
- If there are multiple frames to be included in the result, use the set_frame function overloaded with the singledispatchmethod.
  - like this https://github.com/tier4/driving_log_replayer/pull/228/files#r1315366146

### CODE REVIEW

File: driving_log_replayer/scripts/localization_evaluator_node.py [hunk 1]
```diff
@@ -205,7 +195,8 @@ def add_convergence_frame(
         self.update()
         return msg_lateral_dist
 
-    def add_ndt_availability_frame(self, msg: DiagnosticArray):
+    @set_frame.register
```
Comment (hayato-m126) [inline driving_log_replayer/scripts/localization_evaluator_node.py:198]: set_frame overload

**Comment by kminoda** [comment]
@hayato-m126 Thank you!!! Is it OK if I review this after Thursday, since I am at Shiojiri now?

**Comment by hayato-m126** [comment]
@kminoda 
No problem. Thank you for quick reply.

**Comment by hayato-m126** [comment]
@kminoda 
Friendly ping.
Can you please review this PR?

**Review by kminoda — APPROVED** [review]
Im sorry I failed to reply.
LGTM

**Pull request merged**